ansible/roles/couchdb/tasks/deploy.yml

```diff
@@ -9,17 +9,23 @@
   fail: msg="The db provider in your {{ inventory_dir }}/group_vars/all is {{ db_provider }}, it has to be CouchDB, pls double check"
   when: db_provider != "CouchDB"
 
-- name: check for persistent disk
-  shell: df -h
-  register: disk_status
-  when: block_device is defined
-
 - name: "set the volume_dir"
-  vars:
-    instance: "{{instances | selectattr('name', 'equalto', 'db') | list | first}}"
   set_fact:
-    volume_dir: "{{ instance.volume.fsmount | default( '/mnt/' + group_names|first, true ) }}:/usr/local/var/lib/couchdb"
-  when: (block_device is defined) and (block_device in disk_status.stdout)
+    volume_dir: "{% if db_datadir is defined and db_confdir is defined  %}[ '{{ db_confdir }}:/opt/couchdb/etc', '{{ db_datadir }}:/opt/couchdb/data' ]{% elif db_datadir is undefined  %}{{ db_confdir }}:/opt/couchdb/etca{% else %}{{ db_datadir }}:/opt/couchdb/data{%- endif %}"
```

Comment:
In the past, the persistent part was defined with the block_device.
Which changes have to be made, if someone already used this?
Does he only need to set db_datadir to the same like block_device before?

Reply:
Yes, they can just specify the directory in `db_datadir` or `db_confdir`.

And I think, as we are using `apache/couchdb`, `block_device` is no more working.
Because data directory of CouchDB is different; `/usr/local/var/lib/couchdb`.